graphql-kotlin-federation/src/test/kotlin/com/expediagroup/graphql/federation/validation/ValidateKeySetFieldKtTest.kt

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2020 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.federation.validation
+
+import graphql.Scalars.GraphQLString
+import graphql.schema.GraphQLDirective
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLInterfaceType
+import graphql.schema.GraphQLList
+import graphql.schema.GraphQLTypeReference
+import graphql.schema.GraphQLUnionType
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+
+class ValidateKeySetFieldKtTest {
+
+    private val externalDirective: GraphQLDirective = GraphQLDirective.newDirective()
+        .name("external")
+        .build()
+
+    @Test
+    fun `returns an error on null targetField`() {
+        val errors = mutableListOf<String>()
+        validateKeySetField(
+            targetField = null,
+            extendedType = false,
+            errors = errors,
+            validatedDirective = "foo")
+
+        assertEquals(1, errors.size)
+        assertEquals("foo specifies invalid field set - field set specifies fields that do not exist", errors.first())
+    }
+
+    /**
+     * type Parent @extends {
+     *   foo: String
+     * }
+     */
+    @Test
+    fun `returns an error on extended type without external directive`() {
```

Comment:
i think this test is somewhat weird - i.e. i believe the original test was verifying that key for extended type is referencing local value

```
type Parent @key(fields: "foo") @extends {
  foo: String
}

Reply:
This is validating this conditional check: https://github.com/ExpediaGroup/graphql-kotlin/pull/540/files#diff-25f40334e67210df324a96883dcffa2cR29